src/main/java/com/scottlogic/filter/FilterOR.java

```diff
@@ -0,0 +1,33 @@
+package com.scottlogic.filter;
+
+import com.scottlogic.UserPost;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class FilterOR implements Filter {
+
+    private Filter firstFilter;
+    private Filter secondFilter;
+
+    public FilterOR(Filter filter1, Filter filter2) {
+        firstFilter = filter1;
+        secondFilter = filter2;
+    }
+
+    @Override
+    public List<UserPost> filter(List<UserPost> inputList) {
+        if (inputList == null) {
+            return Arrays.asList();
+        }
+        List<UserPost> outputList1 = firstFilter.filter(inputList);
+        List<UserPost> outputList2 = secondFilter.filter(inputList);
+
+        List<UserPost> outputList2copy = new ArrayList<>(outputList2);
+        outputList2copy.removeAll(outputList1);
+        outputList1.addAll(outputList2copy);
+
+        return outputList1;
```

Comment:
This is great and works. 

For me, this is slightly more difficult to read than it needs to be as we are; copying the second list, removing duplicate elements from it, then adding those elements to the first list. This is how I would do it if I were say, filtering a pack of cards, but **can we take advantage of [another datatype](https://www.tutorialspoint.com/java/java_set_interface.htm) that doesn't allow duplicates?**

Reply:
Gradually becoming aware of the importance of data types in Java! Implemented a Set() and adjusted tests so that ordering isn't an issue.